File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/generator/ParentGenerator.java
package com.github.tobi812.sprykerplugin.models.generator;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.finder.ClassFinderInterface;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;
import pav.sprykerFileCreator.model.resolver.ClassResolverInterface;

public class ParentGenerator implements ParentGeneratorInterface {

    private ClassResolverInterface classResolver;
    private ClassFinderInterface classFinder;

    public ParentGenerator(
            ClassResolverInterface classResolver,
            ClassFinderInterface classFinder
    ) {
        this.classResolver = classResolver;
        this.classFinder = classFinder;
    }

    @Override
    public PhpClassInterface getParentClass(PhpClassInterface phpClass, ClassDefinitionInterface classDefinition) {

        PhpClassInterface coreClass = this.classResolver.resolveClassBelow(phpClass.getFullQualifiedName());

        if (coreClass == null) {
            coreClass = this.classFinder.findClass(classDefinition.getDefaultParentClass());
        }

        return coreClass;
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/generator/ParentGenerator.kt
package com.github.tobi812.sprykerplugin.models.generator

import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface
import com.github.tobi812.sprykerplugin.models.finder.ClassFinderInterface
import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import com.github.tobi812.sprykerplugin.models.resolver.ClassResolverInterface
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project

class ParentGenerator(
    private val project: Project
) : ParentGeneratorInterface {

    override fun getParentClass(
        phpClass: PhpClassInterface,
        classDefinition: ClassDefinitionInterface
    ): PhpClassInterface? {
        val classResolver = project.service<ClassResolverInterface>()
        var coreClass: PhpClassInterface? = classResolver.resolveClassBelow(phpClass.getFullQualifiedName())
        val defaultParentClass: String? = classDefinition.defaultParentClass
        if (coreClass == null && defaultParentClass != null) {
            coreClass = project.service<ClassFinderInterface>().findClass(defaultParentClass)
        }

        return coreClass
    }
}
